### PR TITLE
Be nicer about where to open the results webview

### PR DESCRIPTION
**Pull request by aeisenberg**
Currently, the webview _always_ opens next to the currently active
editor. This is a pain if you already have 2 columns open since this
means that the webview will open in a third column, which is rarely
what you want.

This change uses a more sophisticated approach to opening the webview:

1. If there is only one column, open webview to the right of it
2. If there are multiple columns and the active editor is _not_ the
   last column, open to the right of the active editor
3. Otherwise open in the first column.

This will avoid opening a new column unless there is only one column
open right now.

There is no native API that vscode exposed to compare column locations,
so this uses the `ViewColumn` api is a slightly non-standard way.

A limitation is that if the last column is empty and the active editor
is to the left of it, then the webview will not be opened there (which
would be nice). Instead, it will be opened in column 1.

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] `@github/docs-content-codeql` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/src/interface.ts [hunk 1]
```diff
@@ -203,6 +204,29 @@ export class InterfaceManager extends DisposableObject {
...
+   * If there is a single view column, then open beside it.
+   * If there are multiple view columns, then open beside the active column,
+   * unless the active editor is the last column. In this case, open in the first column.
+   *
+   * The goal is to avoid opening new columns when there already are two columns open.
+   */
+  private chooseColumnForWebview(): vscode.ViewColumn {
+    // This is not a great way to determine the number of view columns, but I
+    // can't find a vscode API that does it any better.
+    // Here, iterate through all the visible editors and determing the max view column.
```
Comment (shati-patel) [inline extensions/ql-vscode/src/interface.ts:219]: minor :) 
```suggestion
    // Here, iterate through all the visible editors and determine the max view column.
```

**Review by shati-patel — COMMENTED** [review]
I wasn't sure how to review the actual code, but I tried opening webviews in various locations locally, and that worked as expected! 🕸️ 

Could you add a changelog entry for this PR?

**Comment by aeisenberg** [comment]
@shati-patel can I get another look at this one?

File: extensions/ql-vscode/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## [UNRELEASED]
 
+- Avoid creating a third column when opening the results view. The results view will always open to the right of the active editor, unless the active editor is in the rightmost editor column. In that case open in the leftmost column. [#1037](https://github.com/github/vscode-codeql/pull/1037)]
```
Comment (shati-patel) [inline extensions/ql-vscode/CHANGELOG.md:5]: Extra bracket!

```suggestion
- Avoid creating a third column when opening the results view. The results view will always open to the right of the active editor, unless the active editor is in the rightmost editor column. In that case open in the leftmost column. [#1037](https://github.com/github/vscode-codeql/pull/1037)
```

**Review by shati-patel — APPROVED** [review]
Ooops sorry, I completely forgot to approve this 🙈 

Thanks for the ping!!!

File: extensions/ql-vscode/CHANGELOG.md (hunk 2 above)
Comment (aeisenberg) [reply extensions/ql-vscode/CHANGELOG.md:5]: Thanks for your eagle eyes.

**Pull request merged**